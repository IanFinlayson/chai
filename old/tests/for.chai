def main():
    for i in [1 .. 10]:
        print(i, "** 2 =", i ** 2)

    for letter in "booyah!":
        print(letter)
        
    for a in "abc":
        for b in "xyz":
            print(a, b)

    let nums = [3, 17, 41]
    for i in nums:
        for j in nums:
            for k in nums:
                print(i * j * k)


