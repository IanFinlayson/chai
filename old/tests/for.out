1 ** 2 = 1
2 ** 2 = 4
3 ** 2 = 9
4 ** 2 = 16
5 ** 2 = 25
6 ** 2 = 36
7 ** 2 = 49
8 ** 2 = 64
9 ** 2 = 81
10 ** 2 = 100
b
o
o
y
a
h
!
a x
a y
a z
b x
b y
b z
c x
c y
c z
27
153
369
153
867
2091
369
2091
5043
153
867
2091
867
4913
11849
2091
11849
28577
369
2091
5043
2091
11849
28577
5043
28577
68921
